odoo/addons/base/models/ir_actions.py

```diff
@@ -1075,6 +1078,26 @@ def copy_data(self, default=None):
             default['name'] = _('%s (copy)', self.name)
         return super().copy_data(default=default)
 
+    @api.depends("model_name")
+    def _compute_public_method_selection(self):
+        for ba in self:
+            ba.public_method_selection = json.dumps(self._get_model_buttons(ba.model_name))
+
+    @api.model
+    def _get_model_buttons(self, model_name):
+        if not model_name:
+            return []
+
+        ctx = {k: v for k, v in self._context.items() if "_view_ref" not in k}
+        form = self.env[model_name].with_context(ctx).sudo().get_view(view_type="form")["arch"]
+        form = etree.fromstring(form)
+        results = {}
+        for node in form.iter(etree.Element):
+            if node.tag == "field":
+                node.getparent().remove(node)
```

Comment:
What is this for? I suppose to skip a sub-tree by mutating the document.

Reply:
exactly